File: Searching and Sorting Algorithms/codigo_cs/12SelectionSort.cs
using System;

class Test12
{
    static void SelectionSort(int[] arr, int n)
    {
        // write your code here
        for (int i = 0; i < n; i++)
        {
            int min_index = i;
            for (int j = i + 1; j < n; j++)
            {
                if (arr[j] < arr[min_index])
                {
                    min_index = j;
                }
            }

            int temp = arr[min_index];
            arr[min_index] = arr[i];
            arr[i] = temp;
        }

        Console.WriteLine(string.Join(" ", arr));

    }

    public static void Execute12()
    {
        int n = int.Parse(Console.ReadLine());
        int[] arr = Array.ConvertAll(Console.ReadLine().Split(' '), int.Parse);
        SelectionSort(arr, n);
    }
}

File: Searching and Sorting Algorithms/12SelectionSort.py
n=int(input())
arr=list(map(int,input().split()))
for i in range(n):
    m_i=i
    for j in range(i+1,n):
        if arr[j]<arr[m_i]:
            m_i=j
    arr[m_i],arr[i]=arr[i],arr[m_i]

print(*arr)
